ra/registration-authority.go

```diff
@@ -88,14 +88,21 @@ func validateEmail(address string, resolver core.DNSResolver) (rtt time.Duration
 	}
 	splitEmail := strings.SplitN(address, "@", -1)
 	domain := strings.ToLower(splitEmail[len(splitEmail)-1])
-	result, rtt, err := resolver.LookupHost(domain)
-	if err == nil && len(result) == 0 {
-		err = errEmptyDNSResponse
+	var rtt1, rtt2 time.Duration
+	var resultMX []string
+	var resultA []net.IP
+	resultMX, rtt1, err = resolver.LookupMX(domain)
+	if err == nil && len(resultMX) == 0 {
+		resultA, rtt2, err = resolver.LookupHost(domain)
+		if err == nil && len(resultA) == 0 {
+			err = errEmptyDNSResponse
+		}
 	}
 	if err != nil {
 		problem := dns.ProblemDetailsFromDNSError(err)
 		err = core.MalformedRequestError(problem.Detail)
 	}
+	rtt = rtt1 + rtt2
```

Comment:
nit: This will break the `RA.DNS.RTT.A` stat set from the return value of `validateEmail` [here](https://github.com/fanf2/boulder/blob/fanf-1197/ra/registration-authority.go#L234). This isn't really your fault, but it means we should really get #1124 fixed very soon.